xaitk_saliency/utils/bin/det_sal_on_img_dir.py

```diff
@@ -0,0 +1,118 @@
+import click  # type: ignore
+import os
+from PIL import Image  # type: ignore
+import json
+import matplotlib.pyplot as plt  # type: ignore
+import numpy as np
+from typing import TextIO
+import logging
+
+from smqtk_detection.interfaces.detect_image_objects import DetectImageObjects
+from smqtk_core.configuration import from_config_dict
+from smqtk_core.configuration import make_default_config
+
+from xaitk_saliency.interfaces.gen_detector_prop_sal import GenerateDetectorProposalSaliency
+from xaitk_saliency.interfaces.perturb_image import PerturbImage
+from xaitk_saliency.utils.tools.gen_detector_sal import gen_detector_sal
+
+
+@click.command(context_settings={"help_option_names": ['-h', '--help']})
+@click.argument('img_dir', type=click.Path(exists=True, file_okay=False))
+@click.argument('output_dir', type=click.Path(exists=False))
+@click.argument('config_file', type=click.File(mode='r'))
+@click.option('--overlay-image', is_flag=True, help='overlay saliency map on images')
+@click.option('-g', '--generate-config-file', help='write default config to specified file', type=click.File(mode='w'))
+@click.option('--verbose', '-v', count=True, help='print progress messages')
+def det_sal_on_img_dir(
+    img_dir: str,
+    output_dir: str,
+    config_file: TextIO,
+    overlay_image: bool,
+    generate_config_file: TextIO,
+    verbose: bool
+) -> None:
+    """
+    Generate saliency maps for each image in a directory.
+
+    \b
+    IMG_DIR - Directory containg the imgaes.
+    OUTPUT_DIR - Directory to write the saliency maps to.
+    CONFIG_FILE - Configuration file for the DetectImageObjects, PerturbImage,
+        and GenerateDetectorProposalSaliency implementations to use.
+
+    \f
+    :param img_dir: Directory containing the images.
+    :param output_dir: Directory to write the saliency maps to.
+    :param config_file: Config file specifying the ``DetectImageObjects``,
+        ``PerturbImage``, and ``GenerateDetectorProposalSaliency``
+        implementations to use.
+    :param overlay_image: Overlay saliency maps on images. Default is to output
+        write out saliency maps by themselves.
```

Comment:
We should add here that when this is true we gray-scale the image if needed before overlaying a heatmap.